cpp_linter/run.py

```diff
@@ -457,8 +206,10 @@ def verify_files_are_present() -> None:
         file_name = Path(file["filename"])
         if not file_name.exists():
             logger.warning("Could not find %s! Did you checkout the repo?", file_name)
-            logger.info("Downloading file from url: %s", file["raw_url"])
-            Globals.response_buffer = requests.get(file["raw_url"])
+            raw_url = f"https://github.com/{GITHUB_REPOSITORY}/raw/{GITHUB_SHA}/"
+            raw_url += urllib.parse.quote(file["filename"], safe="")
+            logger.info("Downloading file from url: %s", raw_url)
+            Globals.response_buffer = requests.get(raw_url)
             # retain the repo's original structure
             Path.mkdir(file_name.parent, parents=True, exist_ok=True)
             file_name.write_text(Globals.response_buffer.text, encoding="utf-8")
```

Comment:
# This may not work on a submodules' srcs.
This may not be fork friendly in case someone runs the CI on a renamed fork.

>```py
> f"https://github.com/{GITHUB_REPOSITORY}/raw/{GITHUB_SHA}/"
>```

- `GITHUB_REPOSITORY` will always be set to the name of the repo running the CI.
- `GITHUB_SHA` will always be set to the commit specific of the repo running the CI.

By default, we ignore submodules. But if the CI didn't checkout submodules and a submodule was explicitly not ignored, this code will get hit. When we get an incorrect GET response (because the `raw_url` was wrong), then there may be undefined behavior:
>```py
>file_name.write_text(Globals.response_buffer.text, encoding="utf-8")
>```